renovate.json

```diff
@@ -0,0 +1,37 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": ["config:base"],
+  "includePaths": ["packages/create-plugin/templates/**"],
+  "separateMajorMinor": false,
+  "enabledManagers": ["regex"],
+  "customManagers": [
+    {
+      "customType": "regex",
+      "fileMatch": ["_package.json"],
+      "matchStrings": ["\"(?<depName>@grafana/.*)\"[^\"]*:\\s*\"\\^(?<currentValue>.*)\""],
+      "datasourceTemplate": "npm"
+    },
+    {
+      "customType": "regex",
+      "fileMatch": ["plugin.json"],
```

Comment:
I am not 100% sure about the change to `plugin.json` -> `grafanaDependency` -> what you usually want there as a developer is to target the lowest compatible version of grafana to reach wider audience. Always moving it forward to the latest will make the newly scaffolded plugins reach less grafana instances. 

I am not sure there is an automated solution to this - maybe it rather should be (I know renovate probably does not work this way):
- Try to bump dependencies without adjusting the `grafanaDepedency`, if all integration tests pass - great can be auto merged. If not try to increase `grafanaDependency` gradually until tests work...

Reply:
Same concern as above

Reply:
Hmm, although I understand the concern, I am afraid that the safest would be to update `plugin.json->grafanaDependency` and the grafana package dependencies hand-in-hand. The problem is that the package dependencies are `>=x.x.x`, and there can be non-backward-compatible changes in them (even if they are not breaking).

**For example:** there is a new component introduced in `11.1.0`, which doesn't exist in `11.0.0`. We don't use that component in our E2E tests yet, but let's say a plugin dev starts leveraging it. In case we leave `grafanaDependency: >=11.0.0`, but bump the package versions, then it can happen that the plugin breaks runtime.

**Ideas:**
- only do `major` updates for docker images, so when they run the plugin locally via docker-compose they see that it's breaking (and then they can bump the image manually if they want)
- maintain different release channels and branches (e.g. "next" or "canary")

Reply:
> only do major updates for docker images, so when they run the plugin locally via docker-compose they see that it's breaking (and then they can bump the image manually if they want)

Not sure what you mean - you mean set the docker image version (grafana) to `11.0.0` and `grafanaDependency` to `11.1.0`?  Do you mean devs would then manually lower the `grafanaDependency` to 11.0.0 once they see it works?

Reply:
Yeah, or they would increase the docker image version to `11.1.0` as well if they see that it does **not** work and they don't care about wider support. But that's just one idea.

Reply:
> there is a new component introduced in 11.1.0, which doesn't exist in 11.0.0. We don't use that component in our E2E tests yet, but let's say a plugin dev starts leveraging it. In case we leave grafanaDependency: >=11.0.0, but bump the package versions, then it can happen that the plugin breaks runtime.

This is an interesting scenario. What one can do when facing such a scenario is to have two branches of the plugin, where one is compatible with >=11.1.0, and the other branch is compatible with <=11.0.0. I The most common scenario in the wild though, is to switch the rendering depending on the grafana version. E2E tests also supports conditional testing depending on what grafana version is currently under test.

Reply:
I think it makes sense to exclude plugin json grafanaDepedency from the Renovate updates. I'll change grafanaDependency to the last minor of the previous major (>=10.4.0) in this PR.